fastlane/lanes/release.rb

```diff
@@ -118,31 +221,75 @@
+
+      • Current internal release version and build code: #{current_internal_release_version} (#{current_internal_build_code}).
+      • New internal release version and build code: #{new_version} (#{internal_hotfix_build_code(release_version: new_version)}).
+
+      Branching from tag: #{previous_version}
+
+    MESSAGE
+
+    if options[:skip_confirm]
+      UI.message(message)
+    else
+      UI.user_error!('Aborted by user request') unless UI.confirm("#{message}Do you want to continue?")
+    end
+
+    # Check tags
+    UI.user_error!("Version #{new_version} already exists! Abort!") if git_tag_exists(tag: new_version)
+    UI.user_error!("Version #{previous_version} is not tagged! A hotfix branch cannot be created.") unless git_tag_exists(tag: previous_version)
+
+    # Create the hotfix branch
+    UI.message 'Creating hotfix branch...'
+    ensure_git_status_clean
+    Fastlane::Helper::GitHelper.create_branch("release/#{new_version}", from: previous_version)
+    UI.message "Done! New hotfix branch is: #{git_branch}"
+
+    # Bump the hotfix version and build code and write it to the `xcconfig` file
+    UI.message 'Bumping hotfix version and build code...'
+    PUBLIC_VERSION_FILE.write(
+      version_short: new_version,
+      version_long: hotfix_build_code
+    )
+    UI.message "Done! New Release Version: #{current_release_version}. New Build Code: #{current_build_code}"
+
+    # Bump the internal hotfix version and build code and write it to the `xcconfig` file
+    UI.message 'Bumping internal hotfix version and build code...'
+    INTERNAL_VERSION_FILE.write(
+      version_short: new_version,
+      version_long: internal_hotfix_build_code
     )
+    UI.message "Done! New Internal Release Version: #{current_internal_release_version}. New Internal Build Code: #{current_internal_build_code}"
+
+    Fastlane::Helper::Ios::GitHelper.commit_version_bump
   end
 
   # Finalizes a hotfix, by triggering a release build on CI
   #
   # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
   #
   desc 'Performs the final checks and triggers a release build for the hotfix in the current branch'
-  lane :finalize_hotfix_release do |options|
-    ios_finalize_prechecks(options)
+  lane :finalize_hotfix_release do
+    ensure_git_branch(branch: '^release/') # Match branch names that begin with `release/`
+    ensure_git_status_clean
     git_pull
 
-    if prompt_for_confirmation(
-      message: 'Ready to push changes to remote and trigger the release build?',
-      bypass: ENV.fetch('RELEASE_TOOLKIT_SKIP_PUSH_CONFIRM', nil)
-    )
-      push_to_git_remote(tags: false)
-      trigger_release_build
-    else
-      UI.message('Aborting hotfix finalization. See you later.')
-    end
+    UI.message = "Triggering hotfix build for version: #{current_release_version}"
+    trigger_release_build(branch_to_build: "release/#{current_release_version}")
```

Comment:
I initially removed it because some of the other apps didn't have a confirmation flow. I thought it made sense because the `finalize_hotfix_release` only triggers a CI build and nothing else. But on second thought, the confirmation doesn't hurt anything and is good to have in place in case more steps get added to this lane in the future. Updated in https://github.com/wordpress-mobile/WordPress-iOS/commit/21bf9fd0724eebb10be848aa26f5016401f4db89